oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/process/ProcessTraffic.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.skywalking.oap.server.core.storage.StorageHashMapBuilder;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.skywalking.oap.server.core.source.DefaultScopeDefine.PROCESS;
+
+@Stream(name = ProcessTraffic.INDEX_NAME, scopeId = PROCESS,
+    builder = ProcessTraffic.Builder.class, processor = MetricsStreamProcessor.class)
+@MetricsExtension(supportDownSampling = false, supportUpdate = true)
+@EqualsAndHashCode(of = {
+        "instanceId",
+        "name",
+})
+public class ProcessTraffic extends Metrics {
+    public static final String INDEX_NAME = "process_traffic";
+    public static final String SERVICE_ID = "service_id";
+    public static final String INSTANCE_ID = "instance_id";
+    public static final String NAME = "name";
+    public static final String LAYER = "layer";
+    public static final String AGENT_ID = "agent_id";
+    public static final String PROPERTIES = "properties";
+    public static final String LAST_PING_TIME_BUCKET = "last_ping";
+    public static final String DETECT_TYPE = "detect_type";
+
+    private static final Gson GSON = new Gson();
+
+    @Setter
+    @Getter
+    @Column(columnName = SERVICE_ID)
+    private String serviceId;
+
+    @Setter
+    @Getter
+    @Column(columnName = INSTANCE_ID, length = 600)
+    private String instanceId;
+
+    @Setter
+    @Getter
+    @Column(columnName = NAME, length = 500)
+    private String name;
+
+    @Setter
+    @Getter
+    @Column(columnName = LAYER)
+    private int layer = Layer.UNDEFINED.value();
+
+    @Setter
+    @Getter
+    @Column(columnName = LAST_PING_TIME_BUCKET)
+    private long lastPingTimestamp;
+
+    @Setter
+    @Getter
+    @Column(columnName = DETECT_TYPE)
+    private int detectType = ProcessDetectType.UNDEFINED.value();
+
+    @Setter
+    @Getter
+    @Column(columnName = AGENT_ID, length = 500)
+    private String agentId;
+
+    @Setter
+    @Getter
+    @Column(columnName = PROPERTIES, storageOnly = true, length = 50000)
+    private JsonObject properties;
+
+    @Override
+    public boolean combine(Metrics metrics) {
+        final ProcessTraffic processTraffic = (ProcessTraffic) metrics;
+        this.lastPingTimestamp = processTraffic.getLastPingTimestamp();
+        if (processTraffic.getDetectType() > 0) {
+            this.setDetectType(processTraffic.getDetectType());
+        }
+        if (StringUtil.isNotBlank(processTraffic.getAgentId())) {
+            this.agentId = processTraffic.getAgentId();
+        }
+        if (processTraffic.getProperties() != null && processTraffic.getProperties().size() > 0) {
+            this.properties = processTraffic.getProperties();
+        }
+        if (processTraffic.getDetectType() > 0) {
+            this.detectType = processTraffic.getDetectType();
+        }
```

Comment:
Duplicate code logic

```suggestion
        if (StringUtil.isNotBlank(processTraffic.getAgentId())) {
            this.agentId = processTraffic.getAgentId();
        }
        if (processTraffic.getProperties() != null && processTraffic.getProperties().size() > 0) {
            this.properties = processTraffic.getProperties();
        }
        if (processTraffic.getDetectType() > 0) {
            this.detectType = processTraffic.getDetectType();
        }
```